### ISSUE TITLE
Private methods accessed using bracket notation are reported as 'unused'

### ISSUE THREAD

**Issue by kpreisser** [Microsoft/TypeScript#14186]
**TypeScript Version:**  Version 2.3.0-dev.20170220

**Code**

Compile with `strictNullChecks`, `noImplicitAny`, `noUnusedLocals`
```ts
class A {
    private abc(): number {
        return 1;
    }

    public callMethods() {
        let n = this["abc"]();
        console.log(n);
    }
}
```

**Expected behavior:**

No error.

**Actual behavior:**
```
C:\Users\User\Desktop\tstest>tsc --strictNullChecks --noImplicitAny -noUnusedLocals file1.ts
file1.ts(2,13): error TS6133: 'abc' is declared but never used.
```

When hovering over `n`, the IDE displays `let n: number` so the compiler should know that `abc` is actually used in this case.

Thanks!

**Comment by DanielRosenwasser** [comment]
Thanks for the report, but looks like a duplicate of #14137.
